overlord/snapstate/snapmgr.go

```diff
@@ -389,24 +399,24 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 			panic("internal error: invalid local revision built: " + revision.String())
 		}
 		snapst.LocalRevision = revision
-		ss.Revision = revision
-		snapst.Candidate = &snap.SideInfo{Revision: ss.Revision}
+		ss.SideInfo.Revision = revision
+		snapst.Candidate = &snap.SideInfo{Revision: ss.Revision()}
```

Comment:
I see, all the InstallPath tests are still like:

	c.Assert(snapst.Sequence[0], DeepEquals, &snap.SideInfo{
		RealName: "",

etc